### PR TITLE
refactor(YouTube Music): Rename `Hide music video ads` to `Hide video ads` and add patch description

### LINKED ISSUE #3378: bug(YouTube Music - Hide music video ads): Rename patch name and add description
### Bug description

Currently, Hide music video ads is a not an appropriate name as it not only hides music video ads, but song and podcast ads within YT Music as well.

### Error logs

_No response_

### Solution

Rename the patch to `Hide media ads` and the patch description to "Allows listening to music, podcasts and videos without ads." or "Hides ads played when listening to music, podcasts and videos."

### Additional context

_No response_

### Acknowledgements

- [X] I have checked all open and closed bug reports and this is not a duplicate.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

### PR DESCRIPTION
**Pull request by ILoveOpenSourceApplications**
Closes #3378

### CODE REVIEW

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Please dump the API

File: src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt [hunk 1]
```diff
@@ -6,11 +6,12 @@ import app.revanced.patcher.patch.BytecodePatch
 import app.revanced.patcher.patch.annotation.CompatiblePackage
 import app.revanced.patcher.patch.annotation.Patch
 import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
-import app.revanced.patches.music.ad.video.fingerprints.ShowMusicVideoAdsParentFingerprint
+import app.revanced.patches.music.ad.video.fingerprints.ShowMultimediaAdsParentFingerprint
 import app.revanced.util.exception
 
 @Patch(
-    name = "Hide music video ads",
+    name = "Hide multimedia ads",
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: Why not just name the patch "Hide video ads"?

Comment (ILoveOpenSourceApplications) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: The ad types maybe of video format, but they're being shown when playing different instances. So, the name `Hide video ads` will not be appropriate.

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: > The ad types maybe of video format

What do you mean "maybe"?

>  they're being shown when playing different instances

What do you mean instances.

**Comment by ILoveOpenSourceApplications** [comment]
> Please dump the API

How do I do that?

**Comment by oSumAtrIX** [comment]
You have to run the dumpAPI task via gradle: `gradlew apiDump`.

File: src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt (hunk 1 above)
Comment (ILoveOpenSourceApplications) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: Does YT Music only show video ads across all 3 formats of content?

Comment (ILoveOpenSourceApplications) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: Multimedia streaming ads is a suitable term to encompass both audio and display ads within a streaming context.
This term effectively conveys the idea that the ads utilize multiple media formats (audio and visual) to reach the audience on a streaming platform.

Comment (ILoveOpenSourceApplications) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt:13]: From my short and impatient testing, I'm only coming across ads which are in  video formats when patching without `Hide music video ads`.  May I ask whether this patch also removes the promotion  ads for YT premium between each tracks or not?

Comment (ILoveOpenSourceApplications) [reply src/main/kotlin/app/revanced/patches/music/ad/video/HideMultimediaAds.kt]: I'm only seeing video ads and hence renamed the patch accordingly.

File: api/revanced-patches.api [hunk 2]
```diff
@@ -343,14 +343,14 @@ public final class app/revanced/patches/moneymanager/UnlockProPatch : app/revanc
 	public synthetic fun execute (Lapp/revanced/patcher/data/Context;)V
 }
 
-public final class app/revanced/patches/music/ad/video/HideMusicVideoAds : app/revanced/patcher/patch/BytecodePatch {
```
Comment (oSumAtrIX) [inline api/revanced-patches.api:346]: These are breaking changes. If you rename a patch, keep the old patch class and depend on the new renamed patch. Deprecate the old class.

Comment (ILoveOpenSourceApplications) [reply api/revanced-patches.api:346]: I only did the apiDump, like I was previously instructed. What am I supposed to do now?

Comment (ILoveOpenSourceApplications) [reply api/revanced-patches.api:346]: Do I have create `HideMusicVideoAds.kt` for this?

File: src/main/kotlin/app/revanced/patches/music/ad/video/HideVideoAds.kt [hunk 3]
```diff
@@ -25,23 +26,23 @@ import app.revanced.util.exception
...
+            showVideoAdsMethod.addInstruction(0, "const/4 p1, 0x0")
+        } ?: throw ShowVideoAdsParentFingerprint.exception
     }
 }
 
-@Deprecated("This patch class has been renamed to HideMusicVideoAds.")
-object MusicVideoAdsPatch : BytecodePatch(
-    dependencies = setOf(HideMusicVideoAds::class),
+@Deprecated("This patch class has been renamed to HideVideoAds.")
+object VideoAdsPatch : BytecodePatch(
```
Comment (ILoveOpenSourceApplications) [inline src/main/kotlin/app/revanced/patches/music/ad/video/HideVideoAds.kt:44]: Do I simply have to rename here?
```suggestion
object MusicVideoAdsPatch : BytecodePatch(
```

**Comment by ILoveOpenSourceApplications** [comment]
So now everything is looking good?

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**